src/Maker/AdminMaker.php

```diff
@@ -0,0 +1,249 @@
+            $path = $this->projectDirectory.'/config/';
+            $servicesFile = $io->ask(
+                'The services YAML configuration file',
+                is_file($path.'admin.yaml') ? 'admin.yaml' : 'services.yaml',
+                'Sonata\AdminBundle\Command\Validators::validateServicesFile'
+            );
+            $id = $io->ask(
+                'The admin service ID',
+                $this->getAdminServiceId($adminClassBasename),
+                'Sonata\AdminBundle\Command\Validators::validateServiceId'
+            );
+            $input->setOption('services', $servicesFile);
+            $input->setOption('id', $id);
+        } else {
+            $input->setOption('services', false);
+        }
+        $input->setArgument('model', $modelClass);
+        $input->setOption('admin', $adminClassBasename);
+    }
+
+    /**
+     * Configure any library dependencies that your maker requires.
+     *
+     * @param DependencyBuilder $dependencies
+     */
+    public function configureDependencies(DependencyBuilder $dependencies)
+    {
+        // TODO: Implement configureDependencies() method.
+    }
+
+    /**
+     * Called after normal code generation: allows you to do anything.
+     *
+     * @param InputInterface $input
+     * @param ConsoleStyle   $io
+     * @param Generator      $generator
+     */
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator)
+    {
+        $modelClass = Validators::validateClass($input->getArgument('model'));
+        $modelClassBasename = current(\array_slice(explode('\\', $modelClass), -1));
+        $adminClassBasename = $input->getOption('admin') ?: $modelClassBasename.'Admin';
+        $adminClassBasename = Validators::validateAdminClassBasename($adminClassBasename);
+        $managerType = $input->getOption('manager') ?: array_keys($this->availableModelManagers)[0];
+        $modelManager = $this->availableModelManagers[$managerType] ?: current($this->availableModelManagers);
+        $skeletonDirectory = __DIR__.'/../Resources/skeleton';
+
+        $adminClassNameDetails = $generator->createClassNameDetails(
+            $modelClassBasename,
+            'Admin\\',
+            'Admin'
+        );
+
+        $fields = $modelManager->getExportFields($modelClass);
+        $fieldString = '';
+        foreach ($fields as $field) {
+            $fieldString = $fieldString."\t\t\t->add('".$field."')".PHP_EOL;
+        }
+
+        $fieldString .= "\t\t\t";
+
+        $generator->generateClass($adminClassNameDetails->getFullName(),
+            $skeletonDirectory.'/Admin.tpl.php',
+            ['fields' => $fieldString]);
+
+        $generator->writeChanges();
+
+        $io->writeln(sprintf(
+                '%sThe admin class "<info>%s</info>" has been generated under the file "<info>%s</info>".',
+                PHP_EOL,
+                $adminClassNameDetails->getShortName(),
+                $adminClassNameDetails->getFullName())
```

Comment:
the closing parenthesis for `sprintf` should be on line 187, which should look like this: `));`